native-schema-registry/src/main/java/com/amazonaws/services/schemaregistry/deserializer/ProtobufPostprocessor.java

```diff
@@ -0,0 +1,27 @@
+package com.amazonaws.services.schemaregistry.deserializer;
+
+import com.amazonaws.services.schemaregistry.deserializers.protobuf.ProtobufWireFormatDecoder;
+import com.google.protobuf.CodedInputStream;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProtobufPostprocessor {
+    public static byte[] stripMessageIndex(byte[] data) throws IOException {
+        Pair<Integer, CodedInputStream> indexAndStreamPair =
+            ProtobufWireFormatDecoder.getAndRemoveMessageIndex(data);
+
+        CodedInputStream inputStream = indexAndStreamPair.getRight();
+        List<Byte> output = new ArrayList<>();
+        while (!inputStream.isAtEnd()){
+
+            byte b = inputStream.readRawByte();
+            output.add(b);
+        }
+        Byte[] result = new Byte[output.size()];
+        return ArrayUtils.toPrimitive(output.toArray(result));
```

Comment:
Can we avoid this by using https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/CodedInputStream.html#readByteArray-- ?

Reply:
if we use the readByteArray method, then it will insert additional bytes in the very beginning of the array which I believe is the array length. We do not want them so I used this approach